### PR TITLE
Org Search/CS-200

### PR DESCRIPTION
**Pull request by superkat64**
[CS-200](https://fullstacklabs.atlassian.net/browse/CS-200)
**Needs the pagination branch on city-of-sac-client**
## Description

Create new org_search concern to enable searching through organizations for the categories associated with their services 
## Motivation and Context

Needed resources available feature and pagination functionality

## Issue Link
https://fullstacklabs.atlassian.net/browse/CS-200

## How Do I Test This?
Run db migration
Use postman to make a request to
`http://localhost:8080/api/organizations/search?category[]=Not+for+Profit&per_page=2&page=1`
Should return with two organizations.

### CODE REVIEW

File: app/models/concerns/org_search.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          "organizations.id IN (?)",
+          query.joins(services: :categories).where(
+            'categories.name = ?', category_name
+          ).pluck(:id)
+        )
+      end
+      query.distinct
+    end
+
+    def text_search(params = {})
```
Comment (mfpiccolo) [inline app/models/concerns/org_search.rb]: This method can be deleted

File: app/models/concerns/org_search.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      allowed_params(params).reduce(self) do |relation, (scope_name, value)|
+        value.present? ? relation.public_send(scope_name, value) : relation.all
+      end
+    end
+
+    def search_by_location_bounds(params = {})
+      # includes(:locations).where(fancy bounding query)
+    end
+
+    def search(params = {})
```
Comment (mfpiccolo) [inline app/models/concerns/org_search.rb]: This method can be deleted

File: app/models/concerns/org_search.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    def text_search(params = {})
+      return where(nil) if params.blank?
+      allowed_params(params).reduce(self) do |relation, (scope_name, value)|
+        value.present? ? relation.public_send(scope_name, value) : relation.all
+      end
+    end
+
+    def search_by_location_bounds(params = {})
```
Comment (mfpiccolo) [inline app/models/concerns/org_search.rb]: This was a todo but should be deleted in this PR

File: app/models/concerns/org_search.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+require 'email_filter'
+require 'location_filter'
+
+module OrgSearch
+  extend ActiveSupport::Concern
+
+  included do
+    scope :category, (lambda do |category|
+      joins(services: :categories).where(categories: { name: category })
+      # includes(services: :categories).where(categories: { name: category })
```
Comment (mfpiccolo) [inline app/models/concerns/org_search.rb]: Delete this comment

File: app/models/organization.rb [hunk 5]
```diff
@@ -38,8 +36,6 @@ def self.with_locations(ids)
   extend FriendlyId
   friendly_id :name, use: [:history]
 
-  after_save :touch_locations, if: :needs_touch?
```
Comment (mfpiccolo) [inline app/models/organization.rb:41]: Why did we remove this?

File: db/migrate/20171006185226_create_pg_search_documents.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+class CreatePgSearchDocuments < ActiveRecord::Migration
```
Comment (mfpiccolo) [inline db/migrate/20171006185226_create_pg_search_documents.rb]: I don't think we need this in this PR

File: db/structure.sql [hunk 7]
```diff
@@ -472,6 +472,39 @@ ALTER SEQUENCE organizations_id_seq OWNED BY organizations.id;
 
 
 --
+-- Name: pg_search_documents; Type: TABLE; Schema: public; Owner: -
+--
+
+CREATE TABLE pg_search_documents (
```
Comment (mfpiccolo) [inline db/structure.sql]: I don't think we need this in this PR or maybe at all

File: app/models/organization.rb (hunk 5 above)
Comment (superkat64) [reply app/models/organization.rb:41]: I'm not sure, it was your change

Comment (mfpiccolo) [reply app/models/organization.rb:41]: K let's add that back and see.  I think it is needed.

**Review by mfpiccolo — APPROVED** [review]

**Pull request merged**